.buildkite/scripts/upload_firecracker_packages.sh

```diff
@@ -61,9 +61,6 @@ readonly PACKAGES=(
 # This is the list of packages that actually have different shas
 new_packages=()
 
-echo "--- Building packages"
```

Comment:
I'd really like to comment on the file as a whole, but this is what Github gives me....

At this point, this script isn't so much about uploading Firecracker packages as it is about uploading raw artifacts to Cloudsmith with some "input files checksum" logic to work around promotion issues. If we could somehow refine the interface for this script to better highlight that, it would be nice.

In particular, it's now a little odd that we have the Firecracker package paths hard-coded in this script, but also have them very explicitly mentioned in the pipeline file (my comments about those aside); I would expect to either have the paths in the pipeline file and then pass them into this script, or simply assume we're uploading all of `dist` in the pipeline file, and then hard-coding the file names here, rather than hard-coding them in both places, but without an obvious link between them.

The fact that the presence of the artifact metadata JSON files is implicitly assumed is also a little surprising, now that they're not actually generated in the course of running this script.

We could probably address all of this by passing in a path to a directory (e.g., "dist"), and maybe an extension (".tar.gz"); the implication being "we're going to upload all the .tar.gz files in dist to Cloudsmith; we're also assuming that each .tar.gz file also has a corresponding artifact metadata file". This could also be spelled out in a top-level comment in this script (that also needs to be updated anyway, since it talks about actually making the artifacts in this file).

I think that would really focus what the core work of this script is, and also make it easier to reuse for other "raw" packages in the future, should that be required.

Reply:
I agree with all of the above, and I'll try to address that tonight